lib/core/src/Cardano/Wallet/Primitive/Types/MinimumUTxO.hs

```diff
@@ -0,0 +1,142 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+
+-- |
+-- Copyright: © 2022 IOHK
+-- License: Apache-2.0
+--
+-- Defines the 'MinimumUTxO' type and related functions.
+--
+module Cardano.Wallet.Primitive.Types.MinimumUTxO
+    (
+    -- * Types
+      MinimumUTxO (..)
+    , MinimumUTxOForShelleyBasedEra (..)
+
+    -- * Constructor functions
+    , minimumUTxONone
+    , minimumUTxOConstant
+    , minimumUTxOForShelleyBasedEra
+    )
+    where
+
+import Prelude
+
+import Cardano.Api.Shelley
+    ( ShelleyBasedEra, ShelleyLedgerEra, fromLedgerPParams )
+import Cardano.Ledger.Core
+    ( PParams )
+import Cardano.Wallet.Primitive.Types.Coin
+    ( Coin )
+import Control.DeepSeq
+    ( NFData (..) )
+import Data.Function
+    ( on )
+import Fmt
+    ( Buildable (..), blockListF )
+
+--------------------------------------------------------------------------------
+-- The 'MinimumUTxO' type
+--------------------------------------------------------------------------------
+
+-- | Represents a function for computing minimum UTxO values.
+--
+data MinimumUTxO where
+    MinimumUTxONone
+        :: MinimumUTxO
+        -- ^ Indicates that there is no minimum UTxO value.
+    MinimumUTxOConstant
+        :: Coin
+        -> MinimumUTxO
+        -- ^ Indicates a constant minimum UTxO value.
```

Comment:
You could mention that this is only used for testing.

Reply:
> You could mention that this is only used for testing.

Thanks for this suggestion! I'll add a comment.

Reply:
@Anviking Fixed in 727c7cb95840f0deb99b2570e4fc7df6c8211219.